src/index.ts

```diff
@@ -1,65 +1,108 @@
 import ejs from 'ejs'
-import { helpers, getContext, Property, Parameter } from "@dylibso/xtp-bindgen"
+import { helpers, getContext, ObjectType, EnumType, ArrayType, XtpNormalizedType, MapType, Parameter, Property } from "@dylibso/xtp-bindgen"
 
-function toTypeScriptType(property: Property | Parameter): string {
-  let tp
-  if (property.$ref) {
-    tp = property.$ref.name
-  } else {
-    switch (property.type) {
-      case "string":
-        if (property.format === 'date-time') {
-          tp = 'Date'
-        } else {
-          tp = "string"
-        }
-        break
-      case "integer":
-        if (property.format === 'int64') {
-          throw Error(`We do not support format int64 yet`)
-        } else {
-          tp = "number"
-        }
-        break
-      case "number":
-        tp = "number"
-        break
-      case "boolean":
-        tp = "boolean"
-        break
-      case "object":
-        tp = "any"
-        break
-      case "array":
-        if (!property.items) {
-          tp = 'Array<any>'
-        } else {
-          // TODO this is not quite right to force cast
-          tp = `Array<${toTypeScriptType(property.items as Property)}>`
-        }
-        break
-      case "buffer":
-        tp = "ArrayBufferLike"
-        break
-    }
+function toTypeScriptTypeX(type: XtpNormalizedType): string {
+  // annotate with null if nullable 
+  const nullify = (t: string) => `${t}${type.nullable ? ' | null' : ''}`
+
+  switch (type.kind) {
+    case 'string':
+      return nullify('string')
+    case 'int32':
+    case 'float':
+    case 'double':
+    case 'byte':
+      return nullify('number')
+    case 'date-time':
+      return nullify('Date')
+    case 'boolean':
+      return nullify('boolean')
+    case 'array':
+      const arrayType = type as ArrayType
+      return nullify(`Array<${toTypeScriptTypeX(arrayType.elementType)}>`)
+    case 'buffer':
+      return nullify('ArrayBufferLike')
+    case 'object':
+      const oType = (type as ObjectType)
+      if (oType.properties?.length > 0) {
+        return nullify(oType.name)
+      } else {
+        return nullify('any')
+      }
+    case 'enum':
+      return nullify((type as EnumType).name)
+    case 'map':
+      const { keyType, valueType } = type as MapType
+      return nullify(`Record<${toTypeScriptTypeX(keyType)}, ${toTypeScriptTypeX(valueType)}>`)
+    case 'int64':
+      throw Error(`We do not support format int64 yet`)
```

Comment:
we may want to make this a bigint?

Reply:
Yeah, `bigint` would cover it!

Reply:
I like how (at least in Chrome) throws: 

```
Do not know how to serialize a BigInt
```

😆